src/formatDate.js

```diff
@@ -9,6 +9,57 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const divider = fromFormat[3];
+  const dividerNew = toFormat[3];
+  const parts = date.split(divider);
+
+  const fromFormatMap = {};
+
+  fromFormat.forEach((format, index) => {
+    fromFormatMap[format] = parts[index];
+  });
+
+  const { YYYY, YY, MM, DD } = fromFormatMap;
+
+  let yearPart;
+
+  if (YYYY !== undefined) {
+    yearPart = YYYY;
+  } else if (YY !== undefined) {
+    const shortYear = parseInt(YY, 10);
+
+    yearPart = shortYear < 30 ? `20${YY}` : `19${YY}`;
```

Comment:
The condition to determine the full year from a two-digit year is correct, but consider adding a comment to explain the logic for future reference.